src/runtime/profiler.cpp

```diff
@@ -182,7 +180,7 @@ WEAK int halide_profiler_pipeline_start(void *user_context,
 
     if (!s->started) {
         halide_start_clock(user_context);
-        halide_spawn_thread(sampling_profiler_thread, NULL);
+        s->sampling_thread = halide_spawn_thread(sampling_profiler_thread, NULL);
         s->started = true;
```

Comment:
With this change, I don't think there's a need for the "started" field. It's equivalent to the truth value of sampling_thread.

Reply:
Will remove.